### ISSUE TITLE
TIFF: Allow reading non-normalized/non-clamped values for floating point

### ISSUE THREAD

**Issue by butlnor** [haraldk/TwelveMonkeys#775]
Regarding my later comment in: https://github.com/haraldk/TwelveMonkeys/issues/740#issuecomment-1491668380

The TIFF reader properly reads the floating point values, but unfortunatelly it then clamps the values into the 0..1 range, with `clamp` method in `TIFFImageReader.java`. This is not ok, because TIFFs are not just images, but can also be containers for any data, like spatial terrain elevation. 

Floating point TIFF is normally not meant to be represented directly as grayscale or RGB values. Floating point values are most often mapped into color by some min/max value normalization or some other formula. It's wrong to speculate that 0.0 is always black and 1.0 is zero.

**Comment by haraldk** [comment]
Hi @butlnor,

I understand. However, the `BufferedImage` class represents images, and various operations like `drawImage` or `BufferedImageOp`s will crash/throw exceptions for unnormalized image data. There's also the case with random users of the library who just use `ImageIO.read(anyFile)` and expects everything to "just work" (my experience is that such users don't read the docs, but still complain when "it doesn't work"...). So, to support these use cases, I like to keep the clamping as default. 

However, there are some options to consider for advanced users (none of these are in the library today):

 1. An `ImageReadParam` to control reading/normalization, perhaps similar to (a generalization of?) the `ToneMapper` used for the HDR plugin. The default would still need to be clamped values for the reasons described above, but allowing other normalizations or no normalization at all. 
 2. Using `ImageReader.readRaster(int, ImageReadParam)` instead of `read(int, ImageReadParam)`. This would be analogous to the JPEG plugin, where reading as raster will give you the "raw" YCbCr values stored (not converted to RGB), or HDR plugin, giving the "raw" RGBE values as stored in the file. The TIFF plugin could then just give you the values as stored in the file. 

I think I prefer option 2. it's not a huge task to do.

**Comment by butlnor** [comment]
Ok. So you  are saying the the option 2 is already working as I expect, or you should change something first? I can go with option 2.

**Comment by rfreeman853** [comment]
Another vote for option 2.

**Comment by haraldk** [comment]
Ok, so option 2 it is! 👍🏻 

But no, as I wrote above, none of the options are implemented in the library today. I'll happily accept both PRs and donations. 😉

**Comment by bchapuis** [comment]
@haraldk I tried to address this issue with option 2. I'm not familiar at all with the architecture of TwelveMonkeys, so the proposed change may be a bit naive. However, it effectively grant access to the raw values of the pixels stored in a geotiff with the `Image.readRaster` method.

```
        String urlString = "https://copernicus-dem-90m.s3.eu-central-1.amazonaws.com/Copernicus_DSM_COG_30_N46_00_E008_00_DEM/Copernicus_DSM_COG_30_N46_00_E008_00_DEM.tif";
        URL url = new URL(urlString);

        ImageReader reader = ImageIO.getImageReadersByFormatName("tiff").next();
        reader.setInput(ImageIO.createImageInputStream(url.openStream()));
        Raster raster = reader.readRaster(0, reader.getDefaultReadParam());

        for (int x = 0; x < raster.getWidth(); x++) {
            for (int y = 0; y < raster.getHeight(); y++) {
                System.out.println(raster.getSample(y, y, 0));
            }
        }
````

Thanks a lot for your work on this library.

**Comment by haraldk** [comment]
Fixed via #881
